### PR TITLE
Add a test for null transaction

### PR DESCRIPTION
**Pull request by nijotz**
## Proposed Release Notes
* Added test for null transaction

### CODE REVIEW

File: tests/versioned/koa-router.tap.js [hunk 1]
```diff
@@ -123,6 +123,19 @@ tap.test('koa-router instrumentation', (t) => {
       run()
     })
 
+    t.only('should handle transaction state loss', (t) => {
```
Comment (jgeurts) [inline tests/versioned/koa-router.tap.js]: Should this be `t.test()` instead of `t.only()`?

Comment (nijotz) [reply tests/versioned/koa-router.tap.js]: Good catch! Had that in just for testing. Will push a fix

Comment (nijotz) [reply tests/versioned/koa-router.tap.js]: Fix pushed

**Review by jgeurts — APPROVED** [review]

File: tests/versioned/koa-router.tap.js [hunk 2]
```diff
@@ -123,6 +123,19 @@ tap.test('koa-router instrumentation', (t) => {
...
 
+    t.test('should handle transaction state loss', (t) => {
+      let savedCtx = null
+      router.get('/:any', (ctx) => {
+        savedCtx = ctx
+      })
+      app.use(router.middleware())
+      helper.agent.on('transactionFinished', () => {
+        t.doesNotThrow(() => savedCtx._matchedRoute = 'test')
+        t.end()
```
Comment (michaelgoin) [inline tests/versioned/koa-router.tap.js:134]: I see here we are simulating transaction state loss by writing to `_matchedRoute` once the transaction has completed.

**Review by michaelgoin — APPROVED** [review]
👍
